packages/codegen-ui/lib/validation-helper.ts

```diff
@@ -58,20 +58,64 @@ const studioComponentChildSchema: any = yup.object({
   actions: yup.object().nullable(),
 });
 
-const studioComponentSchema = yup.object({
-  name: alphaNumString().nullable(),
-  id: yup.string().nullable(),
-  sourceId: yup.string().nullable(),
-  componentType: alphaNumNoLeadingNumberString().required(),
-  properties: yup.lazy((value) => propertiesSchema(value).required()),
-  children: yup.array(studioComponentChildSchema).nullable(),
-  figmaMetadata: yup.object().nullable(),
-  variants: yup.array().nullable(),
-  overrides: yup.object().nullable(),
-  bindingProperties: yup.object().nullable(),
-  collectionProperties: yup.object().nullable(),
-  actions: yup.object().nullable(),
-});
+const studioComponentSchema = yup
+  .object({
+    name: alphaNumString().nullable(),
+    id: yup.string().nullable(),
+    sourceId: yup.string().nullable(),
+    componentType: alphaNumNoLeadingNumberString().required(),
+    properties: yup.lazy((value) => propertiesSchema(value).required()),
+    children: yup.array(studioComponentChildSchema).nullable(),
+    figmaMetadata: yup.object().nullable(),
+    variants: yup.array().nullable(),
+    overrides: yup.object().nullable(),
+    bindingProperties: yup.object().nullable(),
+    collectionProperties: yup.object().nullable(),
+    actions: yup.object().nullable(),
+  })
+  // eslint-disable-next-line func-names
+  .test('unique-component-names', 'All component names must be unique', function (value) {
+    const { path, createError } = this;
+    const { duplicateNames } = findCollidingNames(value);
+    if (duplicateNames.size > 0) {
+      return createError({
+        path,
+        message: `Duplicate names are not allowed within a component, found ${JSON.stringify([...duplicateNames])}`,
```

Comment:
nit: maybe change to `... found duplicates of ['Comp1', 'Comp2']`. At first when I was reading the snapshotted error I was confused because I thought it was saying "DeepComp" and "MyComp" were the same component name.

Reply:
Makes sense, I'll update the error verbiage a bit.